lib/bolt/puppetdb/client.rb

```diff
@@ -42,6 +42,22 @@ def facts_for_node(certnames)
         end
       end
 
+      def fact_values(certnames, facts = [])
+        return [] if certnames.empty? || certnames.nil?
+
+        certnames.uniq!
+        name_query = certnames.map { |c| ["=", "certname", c] }
+        name_query.insert(0, "or")
+
+        facts_query = facts.map { |f| ["=", "path", f] } unless facts.empty?
+        facts_query.insert(0, "or") unless facts.empty?
+
+        query = "[\"and\", #{name_query}, #{facts_query}]"
```

Comment:
We should continue to operate on data structures and then `.to_json` the data instead of building a string.